storage/remote/write_handler.go

```diff
@@ -479,55 +484,90 @@ func (h *writeHandler) appendV2(app storage.Appender, req *writev2.Request, rs *
 	return samplesWithoutMetadata, http.StatusBadRequest, errors.Join(badRequestErrs...)
 }
 
+type OTLPOptions struct {
+	// Convert delta samples to their cumulative equivalent by aggregating in-memory
+	ConvertDelta bool
+}
+
 // NewOTLPWriteHandler creates a http.Handler that accepts OTLP write requests and
 // writes them to the provided appendable.
-func NewOTLPWriteHandler(logger *slog.Logger, appendable storage.Appendable, configFunc func() config.Config) http.Handler {
-	rwHandler := &writeHandler{
-		logger:     logger,
-		appendable: appendable,
+func NewOTLPWriteHandler(logger *slog.Logger, reg prometheus.Registerer, appendable storage.Appendable, configFunc func() config.Config, opts OTLPOptions) http.Handler {
+	ex := &rwExporter{
+		writeHandler: &writeHandler{
+			logger:     logger,
+			appendable: appendable,
+		},
+		config: configFunc,
+	}
+
+	steps := []otel.Step{otel.Sink(ex)}
+	if opts.ConvertDelta {
+		steps = []otel.Step{
+			otel.Processor(deltatocumulative.NewFactory(), nil),
+			otel.Sink(ex),
+		}
 	}
 
-	return &otlpWriteHandler{
-		logger:     logger,
-		rwHandler:  rwHandler,
-		configFunc: configFunc,
+	pl, err := otel.Use(reg, steps...)
+	if err != nil {
+		panic(err) // TODO: how to handle?
```

Comment:
Discussed this in a call:

Let's leave the panic here, but update the comment. The only way we could get an error is if the deltatocumulative component receives a bad config, and we are not exposing that config to the user yet.